src/site/stages/build/tome-sync-page-builder.js

```diff
@@ -1,30 +1,27 @@
-const {
-  getModifiedEntity,
-  toId,
-  readEntity,
-  readAllNodeNames,
-} = require('./page-builder/helpers');
+// Dependencies
+const { get } = require('lodash');
+// Relative
+const { getModifiedEntity, toId } = require('./page-builder/helpers');
 
 /**
  * Takes an entity type and uuid, reads the corresponding file,
  * searches for references to other entities, and replaces the
  * references with the contents of those entities recursively.
  *
- * @param {String} entityType - The type of entity; corresponds to the
- *                              name of the file.
- * @param {String} uuid - The UUID of the entity; corresponds to the
- *                        name of the file.
- * @param {Array<String>} parents - A list of parent entities; used to
- *                                  avoid circular references.
+ * @param {Object} entity - The entity object.
  *
  * @return {Object} - The entity with all the references filled in with
  *                    the body of the referenced entities.
  */
-const assembleEntityTree = (entityType, uuid, parents = []) => {
+const assembleEntityTree = (entity, parents = []) => {
+  // Derive entity properties.
+  const targetID = get(entity, 'type[0].target_id');
+  const uuid = get(entity, 'uuid[0].value');
```

Comment:
Previously what would happen if I read the code correctly is it would skip over this line: `if (parents.includes(toId(targetID, uuid))) {` since the `parents` likely does not include a string `undefined.undefined`. Then later in the code (on line 50) it may try to add `undefined.undefined` to the `parents` array for the `modifiedEntry`.

Reply:
For the record: @kelsonic and I looked over this and decided to move forward with merging this PR and writing up a ticket to address this issue separately so we can avoid as many merge conflicts as we can manage.